### ISSUE TITLE
Some async comprehensions are reported incorrectly

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1863]
Currently our `InconsistentComprehensionViolation` fails on some `async` comprehensions. 
We need to fix that!

Reproduction:

```
async def wrapper():
    (
        some(number)
        async for number in matrix
        if number > 0
    )
```

Result:

```
  4:15     WPS361 Found an inconsistently structured comprehension
  async for number in matrix
        ^
```

**Comment by sobolevn** [comment]
Moreover, this code also triggers false-negative:

```python
comp = [
    [a + 1 for a in b]
    for b in some if b > 0
]
```

Correct version:

```python
comp = [
    [a + 1 for a in b]
    for b in some 
    if b > 0
]
```

The funny thing is that 

```python
comp = [
    some(b)
    for b in some 
    if b > 0
]
```

works correctly in both cases.

**Comment by sobolevn** [comment]
We also need to test comprehensions with multiple `if`s.
